src/test/sql/sqlite_testrunner/sqlite_testrunner.cpp

```diff
@@ -141,20 +133,25 @@ TEST_P(SQLiteTestRunner, CompareToSQLite) {
     lqp_translator = std::make_shared<LQPTranslator>();
   }
 
-  SCOPED_TRACE("SQLite " + query + (use_jit ? " with JIT" : " without JIT"));
+  SCOPED_TRACE("Query '" + sql + "'" + (use_jit ? " with JIT" : " without JIT") + " with encoding " +
+               encoding_type_to_string.left.at(encoding_type));
 
-  auto sql_pipeline = SQLPipelineBuilder{query}.with_lqp_translator(lqp_translator).create_pipeline();
+  auto sql_pipeline = SQLPipelineBuilder{sql}.with_lqp_translator(lqp_translator).create_pipeline();
 
-  const auto& result_table = sql_pipeline.get_result_table();
+  const auto result_table = sql_pipeline.get_result_table();
 
+  // Mark modified tables as dirty. Doing this AFTER executing the query will potentially create confusing test results
+  // after a legimitally failing test has failed. This is since the query might have thrown an exception after modifying
+  // a Table, resulting in this dirty-marking never running
+  // TODO(andybody) Find a solution that doesn't need try/catch in hyrise code
```

Comment:
which try/catch are you referring to?

Reply:
none. I am just saying that a potential solution to the aforementioned problem might include a try/catch and that I would find it preferable if we find a solution that doesn't.

Reply:
Do we really need this todo? Worst case is that we refresh a table a couple of times more than necessary, correct?

Reply:
Worst case is that query, e.g., 105 (which includes DELETE or INSERT) fails (i.e., throws an exception) AFTER commiting changes and then every subsequent query will have different results for sqlite and hyrise since they work on different data (because the dirty-marking was skipped due to the aforementioned exception).

Reply:
I don't think I understand the problem with the current solution. Right now, we make sure that if the SQL query contains DML statements, we remove and reinsert the tables before the next test, correct? Isn't that the expected behavior? What am I missing?

Reply:
Let's put this PR on ice and discuss this in person.

Reply:
Okay, we didn't discuss this, but I concluded its an exotic non-issue. Yes, query X failing might lead to queries X+1, X+2, X+n failing as well, but that's acceptable, imo. I removed the possibly confusing comment, let's just run with it as it is.